ntbs-service/DataMigration/NotificationImportService.cs

```diff
@@ -0,0 +1,145 @@
+            {
+                _logger.LogInformation(requestId, $"No notifications found with Id={notificationId}");
+                return null;
+            }
+
+            notifications = LookupAndAssignPostcode(notifications);
+
+            var patientName = notifications.FirstOrDefault().FullName;
+            _logger.LogInformation(requestId, $"{notifications.Count} notifications found to import for {patientName}");
+
+            bool isAnyNotificationInvalid = false;
+            foreach (var notification in notifications)
+            {
+                var linkedNotificationId = notification.LTBRID ?? notification.ETSID;  
+                _logger.LogInformation(requestId, $"Validating notification with Id={linkedNotificationId}");
+
+                var validationErrors = GetValidationErrors(notification);
+                if (validationErrors.Count() == 0)
+                {
+                    _logger.LogInformation(requestId, $"No validation errors found");
+                }
+                else
+                {
+                    isAnyNotificationInvalid = true;
+                    _logger.LogWarning(requestId, $"{validationErrors.Count()} validation errors found:");
+                    foreach (var validationError in validationErrors)
+                    {
+                        _logger.LogWarning(requestId, validationError.ErrorMessage);
+                    }
+                }
+            }
+
+            if (isAnyNotificationInvalid)
+            {
+                _logger.LogInformation(requestId, $"Terminating importing notifications for a patient due to validation errors");
+                return null;
+            }
+
+            _logger.LogInformation(requestId, $"Importing {notifications.Count()} valid notifications");
+            var savedNotifications = await _notificationImportRepository.AddLinkedNotificationsAsync(notifications);
+
+            _logger.LogInformation(requestId, $"Finished importing notification for {patientName}");
+            var newIdsString = string.Join(" ,", savedNotifications.Select(x => x.NotificationId));
+
+            _logger.LogInformation(requestId, $"Imported notifications have following Ids: {newIdsString}");
+            
+            return savedNotifications;
+        }
+
+        private List<Notification> LookupAndAssignPostcode(List<Notification> notifications)
+        {
+            var postcodes = _postcodeService.FindPostcodes(notifications.Select(x => x.PatientDetails.Postcode).ToList());
+            notifications.ForEach(n => {
+                var normalisedPostcode = n.PatientDetails.Postcode?.Replace(" ", "").ToUpper();
+                var lookedUpPostcode = postcodes.FirstOrDefault(p => p.Postcode == normalisedPostcode);
+                n.PatientDetails.PostcodeToLookup = lookedUpPostcode?.Postcode;
+                n.PatientDetails.PostcodeLookup = lookedUpPostcode;
+            });
+            return notifications;

```

Comment:
As per the comment in the place that uses it, I don't think we need a return value here at all, since the result of the function is assinginment in a state-ful model